[FLINK-32556][runtime] Renames contenderID into componentId

XComp opened the pull request:
## What is the purpose of the change

We introduced contenderID in a lot of places with [FLINK-26522](https://issues.apache.org/jira/browse/FLINK-26522). The original multi-component leader election classes of [FLINK-24038](https://issues.apache.org/jira/browse/FLINK-24038) used componentId.

Revisiting that naming made me realize that it's actually wrong. A contender is a specific instance of a component that participates in the leader election. A component, in this sense, is the more abstract concept. contenderID refers to an ID for the specific contender instance but the IDs we're sharing are actually referring to a Flink component and therefore, are the same between different contenders which compete for leadership for the same component. This contradicts the definition of an identifier.

## Brief change log

Renames of all occurrences that match the case-insensitive regex `contender.?id`.

## Verifying this change

Grep through the code for `contender.?id`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

zentol reviewed (commented):
yeessssss; I've been waiting for this change.

There are a bunch of log messages like:
`Revoking leadership to contender {} for {}`
which don't make much sense after this change though.

XComp commented:
I went over the logs containing the term `contender` to address your comment. Additionally, the branch got rebased to most-recent `master` after FLINK-32381 got merged.

XComp commented:
There are two open PRs (#22959, #22919) which I plan to merge to `master` first. I will do a final rebase with the two PRs merged before doing a final pass over the code.

XComp commented:
I rebased to most-recent master after the aforementioned PRs were merged. I fixed the occurrences of `contender.?id` in a separate commit and force-pushed the changes.

zentol commented on flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/ZooKeeperLeaderElectionDriver.java:
```diff
@@ -128,20 +128,20 @@ public boolean hasLeadership() {
 
         LOG.debug(
-                "Write leader information {} for {} to {}.",
+                "Write leader information {} for component '{}' to {}.",
```
some log message use "component" while others use "component id". I'd prefer "component" because it reads much more naturally.
